alg/crc32c_sse42.h

```diff
@@ -0,0 +1,16 @@
+#ifndef _CRC32C_SSE42_H_
+#define _CRC32C_SSE42_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+/**
+ * CRC32C_Update_SSE42(state, buf, len):
+ * Feed ${len} bytes from the buffer ${buf} into the CRC32C whose state is
+ * ${state}.  This implentation uses x86 SSE4.2 instructions, and should only
+ * be used if CPUSUPPORT_X86_CRC32 is defined and cpusupport_x86_crc32 returns
+ * nonzero.  ${len} must be greater than, or equal to, 8.
+ */
+void CRC32C_Update_SSE42(uint32_t *, const uint8_t *, size_t);
```

Comment:
In my testing, there's a nontrivial speedup if we make this

```
uint32_t CRC32C_Update_SSE42(uint32_t, const uint8_t *, size_t);
```

i.e., take the state as an input and output variable rather than passing it through as a pointer.  I suspect the speedup has to do with calling conventions and being able to keep things in registers.

I know I said that normally the _CPUFEATURE variants should be drop-in replacements using the same API, but when there's a performance delta I think it's worth making it slightly different.

Reply:
Hmm, I'm seeing a slight slow-down from this.  I'll re-evaluate when we have the new cpu timing code.

Reply:
I guess it's possible that different compilers have differently quirky optimization behaviours here...